### DISCUSSION TITLE
Follow reaper tempo envelope

### DISCUSSION THREAD

**Discussion by Andreya-Autumn** [helgoboss/helgobox#1141] [Ideas]
Hello. First, congrats on the beta release. This is very nice already, seems super promising. :) 

I see in the UI there's a yellow dev notice about following the reaper tempo. But I think there's no feature request we can vote for yet (feel free to close if I'm wrong). This (especially along with #1043) would be huge for me. Let me give an example of why.  

Let's say I'm playing live together with some singers/instrumentalists. We're working on a song which has arranged "verse" and "chorus"  type parts, and some freer solo parts. The song should also have a long slow buildup of energy.
I've prepared scenes for the various sections. 
...and now we want the tempo to rise from 129 to 136 BPM over the course of 5 minutes, regardless of which scenes I trigger in that order during that time. 

This would be huge for me. Planning *some* events out with envelopes on a timeline, while other things remain non-linear, is one of those things which I've found frustratingly difficult in other "session view" type hosts.

**Comment by Andreya-Autumn** [comment]
Few more notes: 

I figure this is more or less mutually exclusive with #1042. So if that becomes available there'd need to be a toggle to decide who follows whom. 

Following reapers tempo envelope is of course not the only way you could solve this. Playtime could have its own internal tempo change over time too, I guess. Though that would require some more design work. And to me it would make more sense to use Reaper for features for the linear-time events and Playtime for non-linear ones.  

The intention here would be to use MIDI clips primarily. I get that audio clips would present some challenges. That'd be ok for me personally.

**Reply by DerWhistler** [reply]
I think I get what Andreya-Autumn wants and I second this feature as well.
This is of course a different approach (gradually changing).
Mabe possible via ReaLearn and a hires midi controller.
Being able to change the tempo while performing is very important to me as well.
However, no matter how sophisticated it will be in the end, I would insist on having the possibility to preset a tempo for each scene.
Ableton Live does it that way and for what I want it seems to be a good solution.

Thank you Benjamin for making Playtime2 happen. 
I did not like Ableton Live and ditched it. Was (and still do) setting up my live setups with an Octatrack and a Digitakt.
But all that stem export could be so much easier.

**Comment by helgoboss** [comment]
Following the tempo envelope is something I want to research. Definitely not before the final release though.

**Reply by johnsamuelson** [reply]
I check in here every once in a while to see if there's been any movement on this feature, which is kind of a big deal (ahem) for me... is there any news please? I can't see anything obvious on here or in any changelogs.

**Reply by helgoboss** [reply]
No news to be expected at this moment, still need to work on non-REAPER stuff: https://github.com/helgoboss/helgobox/discussions/1524

**Comment by johnsamuelson** [comment]
I think I've voted for this too but yep, just to say this would be tremendously useful for me...
